plugins/geojson/src/main/java/com/mapbox/mapboxsdk/plugins/geojson/GeoJsonPlugin.java

```diff
@@ -0,0 +1,441 @@
+                if (dataModel.getPolygons().size() > 0) {
+                    for (PolyData polyData : dataModel.getPolygons()) {
+                        if (isRandomFillColor)
+                            fillColor = getRandomMaterialColor("400");
+                        if (isRandomStockColor)
+                            stockColor = getRandomMaterialColor("400");
+                        map.addPolygon(new PolygonOptions()
+                                .addAll(polyData.getPoints())
+                                .fillColor(fillColor)
+                                .strokeColor(stockColor));
+                    }
+                }
+            }
+            if (dataModel.getPolylines() != null) {
+                if (dataModel.getPolylines().size() > 0) {
+                    for (PolyData polyData : dataModel.getPolylines()) {
+                        if (isRandomStockColor)
+                            stockColor = getRandomMaterialColor("400");
+                        map.addPolyline(new PolylineOptions()
+                                .addAll(polyData.getPoints())
+                                .color(stockColor)
+                                .width(width));
+                    }
+                }
+            }
+            if (dataModel.getBounds() != null)
+                map.easeCamera(CameraUpdateFactory.newLatLngBounds(dataModel.getBounds(), 50), 500);
+
+            /*
+             * set onMarkerClick Listener and pass properties of marker to the MarkerEventListener
+             */
+            map.setOnMarkerClickListener(new MapboxMap.OnMarkerClickListener() {
+                @Override
+                public boolean onMarkerClick(@NonNull Marker marker) {
+                    MarkerData markerData = markerCollectionHashMap.get(marker);
+                    if (markerEventListener != null)
+                        markerEventListener.onMarkerClickListener(marker, markerData.getProperties());
+                    return false;
+                }
+            });
+        }
+    }
+
+    /**
+     * @param typeColor the type of color as String
+     * @return random color as integer value
+     */
+    private int getRandomMaterialColor(String typeColor) {
+        int returnColor = Color.GRAY;
+        int arrayId = context.getResources().getIdentifier("mdcolor_" + typeColor, "array", context.getPackageName());
+        if (arrayId != 0) {
+            TypedArray colors = context.getResources().obtainTypedArray(arrayId);
+            int index = (int) (Math.random() * colors.length());
+            returnColor = colors.getColor(index, Color.GRAY);
+            colors.recycle();
+        }
+        return returnColor;
+    }
+
+    /**
+     * check validate String of GeoJson before parse it
+     *
+     * @param jsonString the string of GeoJson file
+     * @return if string is valid JSON return true
+     */
+    private boolean isJSONValid(String jsonString) {
```

Comment:
creating an unnecessary json object is not the best way of validating if a string is JSON valid.